### ISSUE TITLE
How do I translate a docker run command to CreateContainerAsync?

### ISSUE THREAD

**Issue by IAmTheBusiness** [dotnet/Docker.DotNet#613]
I'm trying to match the following docker run command to the CreateContainerAsync method:

docker run --name MyEmulator --platform linux/i386 --rm -it -p 3001:3001 -e LANG=C.UTF-8 -v "C:\My\emulator:/data" images/myemulator:version1

Can you give me an example of how to do that? The IPAddress would be 127.0.0.1, as well.

**Comment by HofmeisterAn** [comment]
Take a look into the [IContainerOperationsTests](https://github.com/dotnet/Docker.DotNet/blob/master/test/Docker.DotNet.Tests/IContainerOperationsTests.cs) tests. The `CreateContainerParameters` type holds all properties you need. You can compare the properties to the container create operation in the Docker [docs](https://docs.docker.com/engine/api/v1.42/#tag/Container/operation/ContainerCreate) and set the necessary values.

https://github.com/dotnet/Docker.DotNet/blob/86eb07cabb5d5865755d93a2972c2df010cd32f3/test/Docker.DotNet.Tests/IContainerOperationsTests.cs#L43-L53

**Comment by IAmTheBusiness** [comment]
So far, I have this:

Dictionary<string, IList<PortBinding>> portBindings = new Dictionary<string, IList<PortBinding>>();
        PortBinding portBinding = new PortBinding();
        portBinding.HostPort = "3001";
        portBinding.HostIP = "0.0.0.0";
        List<PortBinding> pList = new List<PortBinding>();
        pList.Add(portBinding);
        portBindings.Add("3001/tcp", pList);
        Dictionary<string, EmptyStruct> volumes = new Dictionary<string, EmptyStruct>();
        volumes.Add("\"C:\\My\\emulator:/data\"", default(EmptyStruct));
        await _dockerClient.Containers.CreateContainerAsync(new CreateContainerParameters()
        {
            Image = "images/myemulator:version1",
            HostConfig = new HostConfig()
            {
                DNS = new[] { "8.8.8.8", "8.8.4.4", "127.0.0.1" },
                PortBindings = portBindings
            },
            Name = "MyEmulator",
            Platform = "linux/i386",
            Volumes = volumes
        });

This gets me almost all the way there. I have a few issues:

In Docker Desktop, when I look at the Mounts under the Inspect tab, instead of seeing the two correct values: 
**/DATA** and **C:\My\emulator**, I see:
**"C:\MY\EMULATOR:/DATA"** (with the quotes) and **/var/lib/docker/volumes/3c77eec55b67e0ccaac4d24a7b2a7222066a76a3e0328384576b83b95a38a3c8/_data**

Additionally, in the Containers list in Docker Desktop, Instead of the the Port(s) value being a hyperlinked **3001:3001** where the hyperlink is **http://localhost:3001**. it's just **3001:3001**. When you waive your mouse over it, it shows **0.0.0.0:3001:3001**

Any idea where I'm going wrong?

**Comment by HofmeisterAn** [comment]
> In Docker Desktop, when I look at the Mounts under the Inspect tab, instead of seeing the two correct values

I think you try to mount a local host path, right? You need to set `CreateContainerParameters.HostConfig.Mounts` instead of `CreateContainerParameters.Volumes`.

> Additionally, in the Containers list in Docker Desktop, Instead of the the Port(s) value being a hyperlinked **3001:3001** where the hyperlink is **http://localhost:3001**. it's just **3001:3001**. When you waive your mouse over it, it shows **0.0.0.0:3001:3001**

Try to set `portBinding.HostIP` to `null`.

**Comment by IAmTheBusiness** [comment]
Instead of `CreateContainerParameters.HostConfig.Mounts`, I used: `CreateContainerParameters.HostConfig.Binds = new[] { @"C:\My\emulator:/data" }`

`null `for `portBinding.HostIP` worked perfectly.

I also added `CreateContainerParameters.Tty = true` to match the `-it` directive in the `docker run` command.

Then everything ran perfectly. It was THEN that the Post(s) value turned to a hyperlink.
